src/context/sessionContext/index.test.tsx

```diff
@@ -262,4 +263,31 @@ describe("SessionContext functionality", () => {
     fireEvent.click(getByText("Logout"));
     await waitFor(() => expect(onError).toHaveBeenCalledTimes(1));
   });
+
+  it("registers a session restore callback if one is provided", async () => {
+    (handleIncomingRedirect as jest.Mock).mockResolvedValueOnce(null);
+    (onSessionRestore as jest.Mock).mockResolvedValueOnce(null);
```

Comment:
`onSessionRestore` does not return anything, so I don't think this is necessary?

```suggestion
```